### PR TITLE
Add support for `screenshareType` and ability to set layout for Archives

### PR DESCRIPTION
**Pull request by dragonmantank**
## Description

Adds support for `screenshareType` layout settings when starting or editing archive and broadcast layouts. As a side effect this also adds support for `layout` to the `Client.start_archive()` method.

This also cleans up some of the documentation around layouts so that developers can better understand the dict structure.

This supersedes #188 (thank you @caffodian for the base PR!)

## Motivation and Context

`screeshareType` was added to the API with 2.19, so this adds support for it in the SDK.

## How Has This Been Tested?

Test scripts and unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: opentok/opentok.py [hunk 1]
```diff
@@ -1262,11 +1282,17 @@ def set_broadcast_layout(self, broadcast_id, layout_type, stylesheet=None):
...
+
+        :param String screenshare_type optional: Layout to use for screenshares. Must
+        set 'layout_type' to 'bestFit'
         """
         payload = {
             "type": layout_type,
         }
 
+        if screenshare_type is not None:
+            payload["screenshareType"] = screenshare_type
```
Comment (caffodian) [inline opentok/opentok.py:1294]: Not sure if it's worth validating in the SDK level, but one gotcha (which ultimately led to me not using this) is that you can't actually use this with any layout_type but stylesheet.

I guess by not enforcing it here, no SDK update is needed _if_ that changes...

File: opentok/opentok.py [hunk 2]
```diff
@@ -535,6 +549,9 @@ def start_archive(
             "resolution": resolution,
         }
 
+        if layout is not None:
```
Comment (slorello89) [inline opentok/opentok.py:552]: I'm assuming this is just meant to be passthrough and the user is meant to build the `layout` object ahead of time? Guess we just never allowed folks to start an archive with a layout before in this SDK?

File: opentok/opentok.py [hunk 3]
```diff
@@ -909,11 +926,17 @@ def set_archive_layout(self, archive_id, layout_type, stylesheet=None):
...
+
+        :param String screenshare_type optional: Layout to use for screenshares. Must
+        set 'layout_type' to 'bestFit'
         """
         payload = {
             "type": layout_type,
         }
 
+        if screenshare_type is not None:
+            payload["screenshareType"] = screenshare_type
```
Comment (slorello89) [inline opentok/opentok.py:938]: More pass-through style stuff here - is I take it there's no pre-flight validation precedent in this SDK - if so I don't think there's anything else to do here - if there is I'd at least check the `layout_type` to ensure that it's `bestFit` as that's the requirement.

File: setup.py [hunk 4]
```diff
@@ -38,6 +38,9 @@ def find_version(*file_paths):
 if sys.version_info[0] < 3 or sys.version_info[1] < 4:
     install_requires.append('enum34')
 
+if sys.version_info[0] < 3:
+    install_requires.append('rsa<=4.0')
```
Comment (slorello89) [inline setup.py:42]: Is this intentional? Seems out of scope

File: tests/test_archive_api.py [hunk 5]
```diff
@@ -517,6 +517,72 @@ def test_start_composed_archive(self):
         expect(archive).to(have_property(u("output_mode"), OutputModes.composed))
         expect(archive).to(have_property(u("url"), None))
 
+    @httpretty.activate
```
Comment (slorello89) [inline tests/test_archive_api.py:520]: Missing updated tests for `set_archive_layout` and `set_broadcast_layout`

**Review by slorello89 — CHANGES_REQUESTED** [review]
Minor questions on the passthroughighness of some of the methods. Missing tests for two updated methods, seeming extraneous update to dependencies.

File: opentok/opentok.py (hunk 3 above)
Comment (dragonmantank) [reply opentok/opentok.py:938]: For right now I'm just leaving it as pass-through, yes.

File: opentok/opentok.py (hunk 1 above)
Comment (dragonmantank) [reply opentok/opentok.py:1294]: It actually can't be used with any `type` other than `bestFit`. It looks like the API is defaulting to that if you only pass a `screenshareType`. `stylesheet` is only used for the custom layout, which is mutually exclusive with `screenshareType` (and the API will return an error that can be caught)

**Review by slorello89 — APPROVED** [review]
LGTM 👍
